Metalama.Framework.Engine/SyntaxGeneration/ContextualSyntaxGenerator.cs

```diff
@@ -39,7 +39,9 @@ static ContextualSyntaxGenerator()
         _roslynSyntaxGenerator = (SyntaxGenerator) field.GetValue( null ).AssertNotNull();
     }
 
-    private readonly ConcurrentDictionary<ITypeSymbol, TypeSyntax> _typeSyntaxCache = new( SymbolEqualityComparer.IncludeNullability );
+    private readonly SyntaxGeneratorForIType _syntaxGeneratorForIType;
+    private readonly ConcurrentDictionary<IType, TypeSyntax> _typeSyntaxCache;
```

Comment:
We cannot store IDeclaration here because the ContextualSyntaxGenerator is not scoped with a CompilationModel but with a Compilation. However we could cache builders since they are not model-specific. Respecting scoping concepts is quite important to reduce complexity.

Reply:
Changed the implementation to use Ref<IType> (required interface changes).